tests/api/goal.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { expect, test } from "@playwright/test";
+import { faker } from "@faker-js/faker";
+
+test.describe(
+  "Goal feature",
+  {
+    tag: ["@Goal"],
+  },
+  () => {
+    test.describe("Tests involving creating new Goals", () => {
+      let createdTaskId: string;
+
+      test("Create new goal", async ({ request }) => {
+        const goalName = faker.person.jobType();
+        const goalDescription = faker.hacker.phrase();
+        const goalColor = faker.color.rgb();
+
+        const newGoalBody = {
+          name: goalName,
+          due_date: 1568036964079,
+          description: goalDescription,
+          multiple_owners: false,
+          owners: [],
+          color: goalColor,
+        };
+
+        const response = await request.post(`/api/v2/team/${process.env.BASE_TEAM_ID}/goal`, { data: newGoalBody });
+        createdTaskId = (await response.json()).goal.id;
+        expect(response).toBeOK();
+      });
+
+      test.afterEach(async ({ request }) => {
+        await request.delete(`https://api.clickup.com/api/v2/goal/${createdTaskId}`);
+      });
+    });
+
+    test.describe("Tests involving deleting Goals", () => {
+      let preparedTaskId: string;
+
+      test.beforeEach(async ({ request }) => {
+        const preparedGoalBody = {
+          name: "",
+          due_date: 1568036964079,
+          description: "",
+          multiple_owners: false,
+          owners: [],
+          color: "",
+        };
+        const response = await request.post(`/api/v2/team/${process.env.BASE_TEAM_ID}/goal`, { data: preparedGoalBody });
+        preparedTaskId = (await response.json()).goal.id;
+      });
+
+      test("Delete a goal", async ({ request }) => {
+        const response = await request.delete(`/api/v2/goal/${preparedTaskId}`);
+        expect(response.status()).toEqual(200);
+      });
+    });
+
+    test.describe("Tests performed on existing Goals", () => {
+      test("Get all non-archived goals", async ({ request }) => {
+        const response = await request.get(`/api/v2/team/${process.env.BASE_TEAM_ID}/goal`);
+        expect(response).toBeOK();
```

Comment:
Tu można też dodać asercję, że pole `golas` w responsie jest niepuste - spodziewamy się, że jakieś goale istnieją, ale jeżeli np. baza ich nie odczyta (co byłoby błędem), to odeśle pustą odpowiedź, ale z kodem `200` - więc test by przeszedł, mimo błędu bazy